airflow/providers/common/sql/operators/sql.py

```diff
@@ -33,7 +33,21 @@
     from airflow.utils.context import Context
 
 
-def parse_boolean(val: str) -> str | bool:
+def _convert_to_float_if_possible(s):
+    """
+    A small helper function to convert a string to a numeric value
+    if appropriate
+
+    :param s: the string to be converted
+    """
+    try:
+        ret = float(s)
```

Comment:
Unnecessary variable, you can return directly (return float(s) or return s in except)